freqtrade/exchange/exchange.py

```diff
@@ -723,9 +723,12 @@ def get_min_pair_stake_amount(self, pair: str, price: float, stoploss: float,
         # The value returned should satisfy both limits: for amount (base currency) and
         # for cost (quote, stake currency), so max() is used here.
         # See also #2575 at github.
-        return self._get_stake_amount_considering_leverage(
-            max(min_stake_amounts) * amount_reserve_percent,
-            leverage or 1.0
+        return min(
+            self._get_stake_amount_considering_leverage(
+                max(min_stake_amounts) * amount_reserve_percent,
+                leverage or 1.0
+            ),
+            min(max_stake_amounts)
```

Comment:
i don't think that'll work (i assume you misunderstood me).

what we try to do is "you can trade between 15$ and 1_000_000$" (assuming that's the maximum) - so the stake-amount must be between these 2.

What this function is now doing:
`min(15, 1_000_000)` - which makes the whole "max" irrelevant - assuming max is always above min - which i would strongly assume).

We'll still need the separate function (or rename this function and return a tuple) - but it's 2 values we'll need to properly have stake-amount between min and max values.